hack/build_channel_catalog.sh

```diff
@@ -58,48 +64,13 @@ else
     bundle_versions_repo_url="https://$BUNDLE_VERSIONS_REPO"
 fi
 
-log "Cloning $BRANCH_CHANNEL branch from $BUNDLE_VERSIONS_REPO into $saas_operator_dir_base"
-git clone --branch "$BRANCH_CHANNEL" "$bundle_versions_repo_url" "$saas_operator_dir_base"
+log "Cloning $BUNDLE_VERSIONS_REPO into $saas_operator_dir_base"
+git clone "$bundle_versions_repo_url" "$saas_operator_dir_base"
```

Comment:
So this is going to start using `master`, which we currently ignore.

- Something normally creates the `staging` and `production` branches. I'm not sure if it's part of the app-interface automation triggered by [this config](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/osd-operators/app.yml) or if it's done by (the first run of) app_sre_build_deploy.sh for a given operator. We should look into that. (Not that having those branches hurts anything; but it's noise.)
- This repository is bootstrapping, so it's a non-issue; but as we look to using this as our example for established projects, we'll need to come up with some kind of transition plan.

Reply:
Some saas-*-bundle repositories have been created in this file: https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/dependencies/gitlab/gitlab.yml. I know nothing about any automation creating branches and directories in those repositories, I don't think there's any. [app_sre_create_image_catalog.sh](https://github.com/openshift/aws-account-operator/blob/master/hack/app_sre_create_image_catalog.sh) assumes both the branch and the directory exist in the saas repo.

I'm fine with staying in `master`. I'm not fine with continuing if there's no specific directory for the operator since the versions repo is going to be shared and we need to differentiate the case of a first run from targetting the wrong versions repository.